src/db.c

```diff
@@ -604,7 +604,6 @@ void flushAllDataAndResetRDB(int flags) {
         rdbSave(server.rdb_filename,rsiptr);
         server.dirty = saved_dirty;
     }
-    server.dirty++;
```

Comment:
@oranagra I'm not sure why server.dirty++ is needed here, at the beginning already server.dirty += emptyDb(-1,flags,NULL);

Reply:
i'm guessing this may be a safety thing, to make sure the FLUSHALL command is replicated to the AOF / Replicas even if the database was empty.
if that's the case, let's add a comment saying that.
but let's be sure..
@redis/core-team @guybe7 anyone has anything to contribute here?

Reply:
@oranagra If the database is empty, can FLUSHALL skip the follow code.
This eliminates the need for server.dirty++.

```c
long long deleted;
deleted = emptyDb(-1,flags,NULL);
if (deleted == 0) return;
server.dirty += deleted;
```

Reply:
@sundb i was thinking that this additional dirty++ is actually needed.
maybe we want to propagate the FLUSHB to replicas and AOF even if the db was already empty.
not sure exactly why we want that, but maybe one thing to consider is modules with out of keyspace data.

another example maybe is something like this:
suppose the user deleted all the keys manually by using the DEL command.
then he issues a manual FLUSHALL.
now, in case snapshots are enabled, we call `rdbSave` to create an empty snapshot.
but without that extra `dirty++`, we won't propagate the FLUSHALL to the replica, and it'll not create an empty persistence file.

bottom line, i tend to think we wanna keep that `dirty++`, and maybe add some comment specifying why.
waiting for more feedback before we proceed.

Reply:
@oranagra It seems to be what you said, it took me a long time to understand it, combined with the call method to figure it out, thanks.
I will continue to see if there are similar problems.

Reply:
@oranagra I add some comment, my english is not good enough, so I quote your reply, I hope you don't mind.